app/src/main/java/com/rajatgoyal/puzzle15/adapter/HighScoresAdapter.java

```diff
@@ -18,44 +22,59 @@
 
 public class HighScoresAdapter extends RecyclerView.Adapter<HighScoresAdapter.HighScoreViewHolder> {
 
-    private ArrayList<HighScore> highScores;
+    private ArrayList<GamePlay> gamePlays;
     private int count;
-    private Context context;
 
+    private Resources resources;
+
+    @NotNull
     @Override
-    public HighScoreViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        context = parent.getContext();
+    public HighScoreViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        resources = context.getResources();
         View view = LayoutInflater.from(context).inflate(R.layout.high_score_list_item, parent, false);
         return new HighScoreViewHolder(view);
     }
 
     @Override
-    public void onBindViewHolder(HighScoreViewHolder holder, int position) {
-        position = count - 1 - position;
-        String movesString = Integer.toString(highScores.get(position).getMoves());
+    public void onBindViewHolder(@NotNull HighScoreViewHolder holder, int position) {
+        GamePlay gamePlay = gamePlays.get(position);
+        String movesString = Integer.toString(gamePlay.getMoves());
+        String scoreString = Integer.toString(gamePlay.getScore());
+
+        holder.score.setText(scoreString);
         holder.moves.setText(movesString);
-        holder.timer.setText(highScores.get(position).getTime().toString());
+        holder.timer.setText(gamePlay.getTime().toString());
+
+        if (position % 2 == 0) {
+            holder.layout.setBackgroundColor(resources.getColor(R.color.colorPrimary));
+        } else {
+            holder.layout.setBackgroundColor(resources.getColor(R.color.light));
+        }
     }
 
     @Override
     public int getItemCount() {
         return count;
```

Comment:
Similarly, here ideally you should return the size of the Array passes, no need for that `count` variable without any purpose